components/core/src/clp_s/ArchiveWriter.hpp

```diff
@@ -95,8 +97,42 @@ class ArchiveWriter {
      * @param key
      * @return the node id
      */
-    int32_t add_node(int parent_node_id, NodeType type, std::string_view const key) {
-        return m_schema_tree.add_node(parent_node_id, type, key);
+    int32_t add_node(int parent_node_id, NodeType type, std::string_view key) {
+        auto const node_id{m_schema_tree.add_node(parent_node_id, type, key)};
+        if (m_matched_timestamp_prefix_node_id == parent_node_id
```

Comment:
Since we only perform operations when `NodeType::Object == type`, can we put it here?